Source/SwiftLintFramework/Rules/Idiomatic/NimbleOperatorRule.swift

```diff
@@ -70,149 +69,156 @@ public struct NimbleOperatorRule: ConfigurationProviderRule, OptInRule, Correcta
-                           location: Location(file: file, characterOffset: $0.location))
+    static func predicateDescription(for node: FunctionCallExprSyntax) -> PredicateDescription? {
+        guard let expectation = node.expectation() else {
+            return nil
         }
-    }
-
-    private func violationMatchesRanges(in file: SwiftLintFile) -> [NSRange] {
-        let contents = file.stringView
-        return rawRegexResults(in: file).filter { range in
-            guard let byteRange = contents.NSRangeToByteRange(start: range.location, length: range.length) else {
-                return false
-            }
-
-            let containsCall = file.structureDictionary.structures(forByteOffset: byteRange.upperBound - 1)
-                .contains(where: { dict -> Bool in
-                    return dict.expressionKind == .call && (dict.name ?? "").starts(with: "expect")
-                })
 
-            return containsCall
-        }
+        return Self.predicatesMapping[expectation.operatorExpr.identifier.text]
     }
+}
 
-    private func rawRegexResults(in file: SwiftLintFile) -> [NSRange] {
-        let operandPattern = "(.(?!expect\\())+?"
+private extension FunctionCallExprSyntax {
+    func expectation() -> Expectation? {
+        guard trailingClosure == nil,
+              argumentList.count == 1,
+              let memberExpr = calledExpression.as(MemberAccessExprSyntax.self),
+              let kind = Expectation.Kind(rawValue: memberExpr.name.text),
+              let baseExpr = memberExpr.base?.as(FunctionCallExprSyntax.self),
+              baseExpr.calledExpression.as(IdentifierExprSyntax.self)?.identifier.text == "expect",
+              let predicateExpr = argumentList.first?.expression.as(FunctionCallExprSyntax.self),
+              let operatorExpr = predicateExpr.calledExpression.as(IdentifierExprSyntax.self) else {
+            return nil
+        }
 
-        let operatorsPattern = "(" + predicatesMapping.map { name, predicateDescription in
-            let argumentsPattern = predicateDescription.arity.hasArguments
-                ? operandPattern
-                : ""
+        let expected = predicateExpr.argumentList.first?.expression
+        return Expectation(kind: kind, baseExpr: baseExpr, operatorExpr: operatorExpr, expected: expected)
+    }
+}
 
-            return "\(name)\\(\(argumentsPattern)\\)"
-        }.joined(separator: "|") + ")"
+private typealias PredicateDescription = (to: String, toNot: String?, arity: Arity)
 
-        let pattern = "expect\\(\(operandPattern)\\)\\.to(Not)?\\(\(operatorsPattern)\\)"
-        let excludingKinds = SyntaxKind.commentKinds
+private enum Arity {
+    case nullary(analogueValue: ExprSyntaxProtocol)
+    case withArguments
 
-        return file.match(pattern: pattern)
-            .filter { _, kinds in
-                excludingKinds.isDisjoint(with: kinds) && kinds.first == .identifier
-            }
-            .map { $0.0 }
+    var hasArguments: Bool {
```

Comment:
This is unused now. https://buildkite.com/swiftlint/swiftlint/builds/4452#01844cf0-961b-40f1-82ef-bc15c829bb9c